### ISSUE TITLE
Debugging fails when targetting .NET Core in Azure Functions

### ISSUE THREAD

**Issue by ToonVanhoutte** [Azure/Azure-Functions#1012]
Just created a new Azure Functions project in Visual Studio 2017, chose the default  HTTP Trigger v2 (.NET Core)  template and pressed F5 to start debugging.  I got this exception:

![image](https://user-images.githubusercontent.com/23122463/47777464-13acf380-dcf5-11e8-8d32-8dfa7a02fd90.png)

My *.csproj file looks like this.  It targets netcoreapp2.1, like @jeffhollan mentionned in the montly webcast.

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <AzureFunctionsVersion>v2</AzureFunctionsVersion>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.*" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
```
If I switch TargetFramework to .NET Standard, everything works as a charm.

```
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <AzureFunctionsVersion>v2</AzureFunctionsVersion>
  </PropertyGroup>
```

Using:

- Visual Studio 2017: 15.8.28010.2048
- Azure Functions Core Tools 2.1.748
- Azure Functions Runtime 2.0.12134.0
- .NET Core SDK 2.1.403
- .NET Core Runtime 2.1.5

Any pointers what's going wrong?
Thx!

**Comment by ioamnesia** [comment]
I had the EXACT same problem. The solution I found was to manually specify the debug settings. Right click your solution and go to Properties, then the Debug Tab. Change the Launch dropdown to Executable, then in the next box (Executable) browse to your installation of dotnet. For me, this was c:\program files\dotnet\dotnet.exe. Finally add this to the Application Arguments: %userprofile%\AppData\Roaming\npm\node_modules\azure-functions-core-tools\bin\func.dll host start

Note, if you're loading the Microsoft.Azure.WebJobs.Extensions.MicrosoftGraph 1.0.0-beta6 read on:

You'll have another problem. On load, it looks for the existence of d:\home\data\byob_graphmap and fails if it cant find it. I did a workaround by creating the folder structure in my c:\temp folder, sharing the temp folder out as "temp" and mapping drive d: to the \\localhost\temp share. Two big bugs to drive you crazy.

**Comment by deanparkr-duplicate** [comment]
Had the same issue here... following @ioamnesia 's steps got things slightly closer, when I debug now I get to this stage "Initializes git repo" and debugging stops. 

![image](https://user-images.githubusercontent.com/718161/47957064-48e06c80-dfa7-11e8-9707-74596ce480ce.png)

It seems like with these kind of issues its not ready for prime time...

**Comment by jeffhollan** [comment]
@ToonVanhoutte What version of `Microsoft.NET.Sdk.Functions` do you have? Make sure that is latest or else netcoreapp2.1 won't work

@deanparkr-duplicate not even sure how you got to this step. Is this from VS?  The screenshot makes it look like you just typed in `func` in the command line

**Comment by deanparkr-duplicate** [comment]
@jeffhollan  I forgot to add "host start" at the end of the application arguments as @ioamnesia suggested. My bad. Now it runs fine with these settings as below.

**However...** it should really be working straight off the bat without this kind of intervention. I have the latest Microsoft.NET.Sdk.Functions. But creating a new project, with those latest functions gets the "A fatal error occurred" message as per the original poster.

Working settings:
![image](https://user-images.githubusercontent.com/718161/47957167-fb64ff00-dfa8-11e8-9d31-99fb20df1fbe.png)

**Comment by jeffhollan** [comment]
@deanparkr-duplicate assuming you have
1. The latest version of the Azure Functions visual studio extension installed
1. The latest templates (when creating a new project you should see if it's refreshing templates or up-to-date)

You shouldn't need to come in here and mess with the startup settings.  The Azure Functions project should correctly.  I'd try to create a new project.  One way to verify is the dropdown should say `v2 (.NET Core)` instead of `v2 (.NET Standard)`.  If you see .NET Core means you have #2 above.

**Comment by deanparkr-duplicate** [comment]
@jeffhollan  You got it..  I downloaded the latest Azure Functions extensions:  https://marketplace.visualstudio.com/items?itemName=VisualStudioWebandAzureTools.AzureFunctionsandWebJobsTools#review-details

Now creating a new project it works fine without that manual intervention. I did not know I needed that extension. When installing VS and selecting the Azure workload it installs the Azure Functions template but it seems this additional extension is needed.
